console/core/src/main/java/org/eclipse/kapua/app/console/core/servlet/FileServlet.java

```diff
@@ -220,9 +223,17 @@ private void doPostCommand(KapuaFormFields kapuaFormFields, HttpServletResponse
         } catch (KapuaIllegalAccessException eiae) {
             resp.sendError(403, eiae.getMessage());
             return;
+        } catch (KapuaDeviceMenagementException edme) {
+            logger.error("Device menagement exception", edme);
+            resp.sendError(404, edme.getMessage());
+            return;
         } catch (Exception e) {
             logger.error("Error sending command", e);
-            throw new ServletException(e);
+            try {
+                throw new KapuaException(KapuaErrorCodes.INTERNAL_ERROR);
```

Comment:
This has some reason to be? 
It seems a rather complicate way to print a stacktrace!

Reply:
@Coduz The problem was that this ServletException was actually showed to the user. So we had something like this:
![screenshot](https://user-images.githubusercontent.com/35954696/48002704-9de9c300-e10c-11e8-9279-8b371d02ff66.png)

My idea was to replace that ServletException with a more generic KapuaException(KapuaErrorCodes.INTERNAL_ERROR) which had to be surrounded with a try/catch block. 
What do you think? Should I change this?

Reply:
By throwing it at line 233 and catching it in 234, and not re-throwing it in the `catch` block causes the exception to be eaten. 

If you wanted to log the exception only, and not returning it to the user, just use

```
new KapuaException(KapuaErrorCodes.INTERNAL_ERROR).printStackTrace();
```
but the exception is already logged at line 231.

If your intention was to throw a KapuaException, the code doesn't do that, and completes the execution of the method with no exception at all in case of an `Exception` thrown by the main method.